src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+  }
+
+  likeBtn.addEventListener("click", () => {
+    if (likeBtn.classList.contains("card__like-button_active")) {
+      api
+        .removeLike(data._id)
+        .then((updatedCard) => {
+          likeBtn.classList.remove("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      api
+        .addLike(data._id)
+        .then((updatedCard) => {
+          likeBtn.classList.add("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  });
+
+  deleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  cardImage.addEventListener("click", () => {
+    previewImage.src = link;
+    previewImage.alt = name;
+    previewCaption.textContent = name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtn = editForm.querySelector(".modal__save-btn");
+  const originalText = saveBtn.textContent;
+  saveBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editNameInput.value,
+      about: editDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      resetValidation(editForm, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      saveBtn.textContent = originalText;
+    });
+}
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+  const saveBtn = addCardForm.querySelector(".modal__save-btn");
+  const originalText = saveBtn.textContent;
+  saveBtn.textContent = "Saving...";
+
+  const newCard = {
+    name: addCardCaptionInput.value,
+    link: addCardLinkInput.value,
+  };
+
+  api
+    .addCard(newCard)
+    .then((cardData) => {
+      cardsList.prepend(createCard(cardData));
+      closeModal(addCardModal);
+      addCardForm.reset();
+      resetValidation(addCardForm, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      saveBtn.textContent = originalText;
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const saveBtn = avatarForm.querySelector(".modal__save-btn");
+  const originalText = saveBtn.textContent;
+  saveBtn.textContent = "Saving...";
+
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      avatarImg.src = data.avatar;
+      closeModal(avatarModal);
+      avatarInput.value = "";
+    })
+    .catch(console.error)
+    .finally(() => {
+      saveBtn.textContent = originalText;
+    });
+}
+
+profileEditBtn.addEventListener("click", () => {
+  editNameInput.value = profileName.textContent;
+  editDescriptionInput.value = profileDescription.textContent;
+  openModal(editModal);
+});
+
+profileAddBtn.addEventListener("click", () => {
+  addCardForm.reset();
```

Comment:
You should ** clear ** the inputs only after **submitting** a card rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it once more, and you’ll have to type the data again.